### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by danepitkin**
Add the following: description, goals, non-goals, and example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,163 @@
 # Substrait
```
Comment (ianmcook) [inline README.md:1]: If we intend to call this package PySubstrait, I think it's best for the heading at the top of the readme to use that name
```suggestion
# PySubstrait
```

File: README.md [hunk 2]
```diff
@@ -1,8 +1,163 @@
...
-A python package for Substrait.
+[Substrait](https://substrait.io) is a cross-language specification for data compute operations.
```
Comment (ianmcook) [inline README.md:2]:
```suggestion

A Python package for [Substrait](https://substrait.io), the cross-language specification for data compute operations.
```

**Comment by ianmcook** [comment]
I'd advocate for naming this "PySubstrait" everywhere, for simplicity. Including in the package names (`pysubstrait`). It makes it easier to refer to it with brevity and precision.

**Comment by danepitkin** [comment]
> I'd advocate for naming this "PySubstrait" everywhere, for simplicity. Including in the package names (`pysubstrait`). It makes it easier to refer to it with brevity and precision.

I'll follow up with a separate PR!

**Pull request merged**